tools/mcsema_disass/binja/cfg.py

```diff
@@ -409,20 +355,23 @@ def recover_function(bv, pb_mod, addr, is_entry=False):
     pb_func.name = func.symbol.name
 
     # Recover all basic blocks
+    var_refs = defaultdict(list)
     for block in func:
         pb_block = add_block(pb_func, block)
 
         # Recover every instruction in the block
-        inst_addr = block.start
-        while inst_addr < block.end:
-            il = func.get_lifted_il_at(inst_addr)
+        for inst in block.disassembly_text:
+            # Skip over anything that isn't an instruction
+            if inst.tokens[0].type != InstructionTextTokenType.InstructionToken:
+                continue
+            il = func.get_lifted_il_at(inst.address)
 
             pb_inst = pb_block.instructions.add()
             recover_inst(bv, pb_block, pb_inst, il)
-            inst_addr += len(pb_inst.bytes)
+            vars.find_stack_var_refs(bv, inst, il, var_refs)
 
     # Recover stack variables
-    recover_stack_vars(pb_func, func)
+    vars.recover_stack_vars(pb_func, func, var_refs)
```

Comment:
This should be conditional on a command-line flag, `--recover-stack-vars` (I think that's the one)